afs/afs-ws/afs-ws-storage/src/main/java/com/powsybl/afs/ws/storage/NodeEventClient.java

```diff
@@ -52,8 +69,52 @@ public void onError(Throwable t) {
         }
     }
 
+
     @OnClose
-    public void onClose(Session session) {
-        LOGGER.trace("Node event websocket session '{}' closed for file system '{}'", session.getId(), fileSystemName);
+    public void onClose(Session session, CloseReason closeReason) {
+
+        if (closeReason.getCloseCode().getCode() == 1006) {
+            URI wsUri = null;
+            if (this.reconnectionInterval > config.getReconnectionMaxInterval()) {
+                LOGGER.warn("Node event websocket session '{}' closed for file system '{}'", session.getId(), fileSystemName);
+                return;
+            }
+            if (counter >= 1) {
+                this.reconnectionInterval = this.reconnectionInterval * config.getReconnectionIntervalMutiplier();
+            } else {
+                this.reconnectionInterval = config.getReconnectionInitialInterval();
+            }
+            WebSocketContainer container;
+            try {
+
+                sleep(reconnectionInterval * 1000L);
+                wsUri = RemoteListenableAppStorage.getWebSocketUri(config.getRestUri());
+                URI endPointUri = URI.create(wsUri + "/messages/" + AfsRestApi.RESOURCE_ROOT + "/" +
+                        AfsRestApi.VERSION + "/node_events/" + fileSystemName);
+                LOGGER.debug("Connecting to node event websocket at {}", endPointUri);
+                container = ContainerProvider.getWebSocketContainer();
+                container.connectToServer(this, endPointUri);
+                resetComptor();
```

Comment:
Rename to resetCounter.

I don’t think counter has to be a static variable: should we have a while loop here that tries to reconnect?!

Reply:
The counter attribute is replaced by a local variable in the method onClose()